featuretools/entityset/entityset.py

```diff
@@ -124,6 +124,11 @@ def __sizeof__(self):
     #     return (EntitySet, serialize.entityset_to_description(self.metadata))
 
     def __eq__(self, other, deep=False):
+        # --> add data_description to equality check???
```

Comment:
We were previously missing attributes of EntitySet that were not `dataframe_dict` or `relationships`, so I figured that `id` and `time_type` made sense to add (though, I do think that if the time type is mismatched, the equality check would fail anyway because the woodwork schemas wouldn't match, but this is a quicker check)

I wasn't sure about whether or not to include `_data_description`, though, because that seems to be something we recalculate at various points, and it mostly a recreation of the EntitySet's information (At least based off of my understanding).

Reply:
I'm thinking we don't need to worry about including `_data_description` (or metadata) in any equality checks. Since the data description is built from the other entityset information as you said, as long as all the other entityset data matches, the data description should as well (as long as it has been generated).

If we were going to check something here, I think it would maybe be best to check the entityset metadata equality as that should regenerate the description if it has been reset and we wouldn't need to worry about handling comparisons with the description possibly in different states between two entitysets. That would require merging in the serialization stuff though.